api/v1beta1/cluster_types.go

```diff
@@ -234,13 +234,19 @@ type MachineHealthCheckTopology struct {
 	MachineHealthCheckClass `json:",inline"`
 }
 
-// ClusterVariable can be used to customize the Cluster through
-// patches. It must comply to the corresponding
-// ClusterClassVariable defined in the ClusterClass.
+// ClusterVariable can be used to customize the Cluster through patches. Each ClusterVariable is associated with a
+// Variable definition in the ClusterClass `status` variables.
```

Comment:
Not sure how much documentation is useful here in terms of the usage of this field. I think it's complex enough that the godoc isn't the right place to describe e.g. that variables can't be used across patches.